docs/content/v0.20/architecture.md

```diff
@@ -1,36 +1,479 @@
+and updates its status. There are two types of status in ManifestWork: the *resourceStatus* 
+tracks the status of each manifest in the ManifestWork, and *conditions* reflects the overall 
+status of the ManifestWork. The work agent checks whether a resource is *Available*, 
+meaning the resource exists on the managed cluster, and *Applied*, meaning the resource 
+defined in ManifestWork has been applied to the managed cluster. To ensure the resources 
+applied by ManifestWork are reliably recorded, the work agent creates an *AppliedManifestWork* 
+on the managed cluster for each ManifestWork as an anchor for resources relating to ManifestWork. 
+When ManifestWork is deleted, the work agent runs a *Foreground* deletion, and that ManifestWork 
+will stay in deleting state until all its related resources have been fully cleaned in the managed 
+cluster.
+
+## OCM Status Add-On Agent
+
+The OCM Status Add-On Agent is a controller that runs alongside the OCM Agent 
+in the managed cluster. Its primary function is to track objects delivered 
+by the work agent and report the full status of those objects back to the ITS. 
+The KubeStellar controller can then use different user-defined summarization 
+policies to report status, such as the `wantSingletonReportedState` policy that reports 
+full status for each deployed object when the workload is delivered only to one 
+cluster. The controller watches *appliedmanifestworks* objects to determine which 
+objects have been delivered through the work agent. It then starts dynamic informers 
+to watch those objects, collect their individual statuses, and report back the status 
+updating *workstatuses* objects in the namespace associated with the WEC in the ITS.
+
+## KubeStellar Controllers Architecture
+
+The KubeStellar controllers architecture is based on common patterns and best 
+practices for Kubernetes controllers, such as the 
+[<u>Kubernetes Sample Controller</u>](https://github.com/kubernetes/sample-controller). 
+A Kubernetes controller uses informers to watch for changes in Kubernetes
+objects, caches to store the objects, event handlers to react to
+events, work queues for parallel processing of tasks, and a reconciler
+to ensure the actual state matches the desired state. However, that
+pattern has been extended to provide the following features:
+
+- Using dynamic informers
+- Starting informers on all API Resources (except some that do not need
+  watching)
+- Informers and Listers references are maintained in a hash map and
+  indexed by GVK (Group, Version, Kind) of the watched objects.
+- Using a common work queue and set of workers, where the key is defined as follows:
+  - Key is a struct instead than a string, and contains the following:
+    - GVK of the informer and lister for the object that generated the
+      event
+    - Structured Name of the object 
+    - For delete event: Shallow copy of the object being deleted. This
```

Comment:
A given finalizer is typically managed by one controller, for the benefit of that controller. Why would it matter here whether an object has any finalizers at all?

Reply:
You are correct it should not matter, as even if there is a finalizer there is no guarantee that that finalizer won't be removed by the owning controller and object will be deleted by the time the worker picks the object ref from the queue.

Reply:
I think GitHub has lost track of where this thread belongs. The same topic is addressed in another thread.